### PR TITLE
Blockly blocks for Motor and Touch Sensor with dummy VM calls

### PR DESCRIPTION
**Pull request by eurocoinsalbum**

### CODE REVIEW

File: src/BlocklyInterface/scripts/DcMotor.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import Blockly from "blockly";
+
+Blockly.defineBlocksWithJsonArray([
```
Comment (sar-k) [inline src/BlocklyInterface/scripts/DcMotor.js:3]: Can this be defined in typescript

File: src/JavascriptVM/JavascriptVM.tsx [hunk 2]
```diff
@@ -88,6 +88,14 @@ export const VMProvider: FunctionComponent = ({ children }) => {
             new BlocklyInterpreter(code, {
               onHighlight: (id) =>
                 dispatch(blocklySlice.actions.highlightBlock({ blockId: id })),
+
+              // dummy implementations
+              onSetDcMotorPower: (port: number, power: number) =>
+                console.log(
```
Comment (sar-k) [inline src/JavascriptVM/JavascriptVM.tsx:94]: should this be debug level

File: src/JavascriptVM/vm.ts [hunk 3]
```diff
@@ -23,8 +25,28 @@ export class BlocklyInterpreter {
...
+      const setDcMotorPower = interpreter.createNativeFunction(
+        (port: number, power: number) => {
+          this.pause = true;
+          callbacks.onSetDcMotorPower(port, power);
+        }
+      );
+
+      const isSensorTouchPushed = interpreter.createNativeFunction(
+        (port: number, power: number) => {
+          this.pause = true;
```
Comment (sar-k) [inline src/JavascriptVM/vm.ts:37]: Would be helpful to have some comments on the purpose of pause flag
The power parameter is not used. Is it needed?

Comment (JamesWP) [reply src/JavascriptVM/vm.ts:37]: i think its not required. can likely be removed

File: src/JavascriptVM/JavascriptVM.tsx (hunk 2 above)
Comment (JamesWP) [reply src/JavascriptVM/JavascriptVM.tsx:94]: this is a placeholder till we have some way of getting these values to the real simulator

Comment (JamesWP) [reply src/JavascriptVM/JavascriptVM.tsx:121]: i removed it

File: src/BlocklyInterface/scripts/DcMotor.js (hunk 1 above)
Comment (JamesWP) [reply src/BlocklyInterface/scripts/DcMotor.js]: there is now a type safe way to do this

**Pull request merged**